DisplayCAL/RealDisplaySizeMM.py

```diff
@@ -80,6 +44,163 @@ def RealDisplaySizeMM(display_no=0):
+        """ICC profile root/output atom for this display."""
+        self.edid = None
+        """128, 256 or 384 bytes of monitor EDID, NULL if none."""
+        self.edid_len = None
+        """128, 256 or 384."""
+
+        # Xrandr stuff - output is connected 1:1 to a display
+        self.crtc = None
+        """Associated crtc."""
+        self.output = None
+        """Associated output."""
+        self.icc_out_atom = None
+        """ICC profile atom for this output."""
+
+    def from_dispwin_data(self, display_info_line):
+        """Parse from dispwin display list data.
+
+        Args:
+            display_info_line (str): The dispwin data line.
+        """
+        display_info_line = display_info_line.strip()
+        description_data = re.findall(
+            rb"[\s\d]+= '(?P<description>.*)'", display_info_line
+        )
+        dispwin_error_message = lang.getstr(
+            "error.generic",
+            (-1, "dispwin returns no usable data while enumerating displays."),
+        )
+        if not description_data:
+            raise ValueError(dispwin_error_message)
+        self.description = description_data[0]
+        match = re.match(
+            rb"[\s]*(?P<id>\d) = '(?P<name>.*), at (?P<x>\d+), (?P<y>[-\d]+), "
+            rb"width (?P<width>\d+), height (?P<height>\d+).*'",
+            display_info_line,
+        )
+        if not match:
+            raise ValueError(dispwin_error_message)
+        groups_dict = match.groupdict()
+        self.monid = int(groups_dict["id"])
+        self.name = groups_dict["name"]
+        x = int(groups_dict["x"])
+        y = int(groups_dict["y"])
+        self.pos = (x, y)
+        width = int(groups_dict["width"])
+        height = int(groups_dict["height"])
+        self.size = (width, height)
+
+    def to_dict(self):
+        """Return a dictionary.
+
+        Returns:
+            dict: The display data as dictionary, matching the previous implementation.
+        """
+        display_dict = {}
+        if self.monid is not None:
+            display_dict["monid"] = self.monid
+        if self.description is not None:
+            display_dict["description"] = self.description
+        if self.name is not None:
+            display_dict["name"] = self.name
+        if self.pos is not None:
+            display_dict["pos"] = self.pos
+        if self.size is not None:
+            display_dict["size"] = self.size
+
+        return display_dict
+
+
+def get_dispwin_output() -> bytes:
+    """Return Argyll dispwin output.
+
+    Returns:
+        bytes: The dispwin output.
+    """
+    dispwin_path = argyll.get_argyll_util("dispwin")
+    if dispwin_path is None:
+        return b""
+
+    if sys.platform == "win32":
+        startupinfo = sp.STARTUPINFO()
+        startupinfo.dwFlags |= sp.STARTF_USESHOWWINDOW
+        startupinfo.wShowWindow = sp.SW_HIDE
+    else:
+        startupinfo = None
+
+    p = subprocess.Popen(
+        [dispwin_path, "-v", "-d0"],
```

Comment:
Hey @eoyilmaz 
that line forces dispwin to use a display that would never be available. With linux (or maybe it is gentoo specific, but I doubt that) the list starts with 1.

see:
```
~/PycharmProjects/displaycal-root 16-get-rid-of-realdisplaysizemm-c-extension 6s
venv ❯ argyll-dispwin -v -d0

Test display patch window, Set Video LUTs, Install profiles, Version 3.2.0
Author: Graeme W. Gill, licensed under the AGPL Version 3
Diagnostic: -d parameter '0' is out of range
usage: dispwin [options] [calfile]
 -v                   Verbose mode
 -display displayname Choose X11 display name
 -d n[,m]             Choose the display n from the following list (default 1)
                      Optionally choose different display m for Video LUT access
    1 = 'Monitor 1, Output DP-4 at 0, 0, width 2560, height 1440'
 -dweb[:port]         Display via web server at port (default 8080)
.....

```

Reply:
Just updated to 3.3.0 to confirm there are no difference between the argyll versions:
```
~/PycharmProjects/displaycal-root 16-get-rid-of-realdisplaysizemm-c-extension* 1m 41s
venv ❯ argyll-dispwin -v -d0

Test display patch window, Set Video LUTs, Install profiles, Version 3.3.0
Author: Graeme W. Gill, licensed under the AGPL Version 3
Diagnostic: -d parameter '0' is out of range
usage: dispwin [options] [calfile]
 -v                   Verbose mode
 -display displayname Choose X11 display name
 -d n[,m]             Choose the display n from the following list (default 1)
                      Optionally choose different display m for Video LUT access
    1 = 'Monitor 1, Output DP-4 at 0, 0, width 2560, height 1440'
 -dweb[:port]         Display via web server at port (default 8080)
.....
```

Reply:
works if set to  ` -d 1`

Reply:
`-d0` is intentional to be honest, it will display the info I require and then exit, otherwise it will start displaying patches.

Reply:
Hey @eoyilmaz sorry for my late reply, but don't you get an eror message on mac os if you use it that way? On linux displaycal won't launch.

Reply:
🤔 it might be related to some other issues, as in my virtualbox Ubuntu 24.10 it starts without any problem.

Reply:
what does the output in the virtual box looks like for:
`argyll-dispwin -v -d0`

Reply:
aah okay, removing the config and restarting now errors in my virtualbox too.

Reply:
![Bildschirmfoto vom 2024-10-22 17-52-26](https://github.com/user-attachments/assets/15b85f03-d3ae-4ed7-9b53-7d00f01c75a9)

Reply:
aah okay it seems that I developed with ArgyllCMS 3.3.0 and ubuntu comes with ArgyllCMS 3.1.0 and the output is slightly different... fixing it quickly.

Reply:
But you are right not the -d 0 is the problem, it's the parsing afterwards. Is the output from dispwin really different on mac os?

Reply:
fixed 👍

Reply:
> Is the output from dispwin really different on mac os?

It is not, but I think it is different between different ArgyllCMS versions.